lib/nats.js

```diff
@@ -1199,6 +1201,31 @@ Client.prototype.request = function(subject, opt_msg, opt_options, callback) {
 };
 
 /**
+ * Publish a message with an implicit inbox listener as the reply. Message is optional.
+ * This should be treated as a subscription. The subscription is auto-cancelled after the
+ * first reply is received or the timeout in millisecond is reached.
+ *
+ * If a timeout is reached, the callback is invoked with
+ * The Subscriber Id is returned.
+ *
+ * @param {String} subject
+ * @param {String} opt_msg
+ * @param {Object} opt_options
+ * @param {Function} callback - can be called with message or NatsError if the request timed out.
+ * @return {Mixed}
+ * @api public
+ */
+Client.prototype.requestOne = function(subject, opt_msg, opt_options, timeout, callback) {
+  opt_options = opt_options || {};
+  opt_options.max = 1;
+  var sid = this.request(subject, opt_msg, opt_options, callback);
+  this.timeout(sid, timeout, 1, function() {
```

Comment:
Ok, I checked the code and I understand now. About the test, though, should it wait for more than next tick to ensure that it is not receiving after timeout? In other words, should you have a node timer that first after more than the timeout and check that count is still 1?